### PR TITLE
Add support for multiple machine types to test_all_sandia.

### PR DESCRIPTION
**Pull request by jgfouca**
Also, begin to add support for white.

### CODE REVIEW

**Comment by jgfouca** [comment]
Hi @crtrott , I finally got some time to work on this. I did some preliminary testing and this change does not appear to break anything on kokkos-dev. These changes also appear to be working as intended on white, but there are many failures in the Kokkos tests which could be due to configuration problems. Please advise on the set of compilers you want to test on white.

File: config/test_all_sandia [hunk 1]
```diff
@@ -6,42 +6,97 @@
...
+    MACHINE=white
+elif [ ! -z "$SEMS_MODULEFILES_ROOT" ]; then
+    MACHINE=sems
+else
+    echo "Unrecognized machine" >&2
+    exit 1
+fi
+
 GCC_BUILD_LIST="OpenMP,Pthread,Serial,OpenMP_Serial,Pthread_Serial"
+IBM_BUILD_LIST="OpenMP,Pthread,Serial,OpenMP_Serial,Pthread_Serial"
```
Comment (jgfouca) [inline config/test_all_sandia]: I guessed that the IBM list would be the same as GCC.

File: config/test_all_sandia [hunk 2]
```diff
@@ -6,42 +6,97 @@
...
 INTEL_BUILD_LIST="OpenMP,Pthread,Serial,OpenMP_Serial,Pthread_Serial"
 CLANG_BUILD_LIST="Pthread,Serial,Pthread_Serial"
 CUDA_BUILD_LIST="Cuda_OpenMP,Cuda_Pthread,Cuda_Serial"
 
 GCC_WARNING_FLAGS="-Wall,-Wshadow,-pedantic,-Werror,-Wsign-compare,-Wtype-limits,-Wignored-qualifiers,-Wempty-body,-Wclobbered,-Wuninitialized"
+IBM_WARNING_FLAGS="-Wall,-Wshadow,-pedantic,-Werror,-Wsign-compare,-Wtype-limits,-Wuninitialized"
```
Comment (jgfouca) [inline config/test_all_sandia]: I guessed that the IBM warnings would be the same as intel.

File: config/test_all_sandia [hunk 3]
```diff
@@ -6,42 +6,97 @@
...
+
+elif [ "$MACHINE" = "white" ]; then
+    SKIP_HWLOC=True
+
+    BASE_MODULE_LIST="<COMPILER_NAME>/<COMPILER_VERSION>"
+    IBM_MODULE_LIST="<COMPILER_NAME>/xl/c/<COMPILER_VERSION>"
+    CUDA_MODULE_LIST="<COMPILER_NAME>/<COMPILER_VERSION>,gcc/4.9.2"
+
+    # Format: (compiler module-list build-list exe-name warning-flag)
+    COMPILERS=("gcc/4.9.2 $BASE_MODULE_LIST $GCC_BUILD_LIST g++ $GCC_WARNING_FLAGS"
```
Comment (jgfouca) [inline config/test_all_sandia]: Guessed a few compilers to test.

File: config/test_all_sandia [hunk 4]
```diff
@@ -6,42 +6,97 @@
...
+               "clang/3.6.1 $BASE_MODULE_LIST $CLANG_BUILD_LIST clang++ $CLANG_WARNING_FLAGS"
+               "cuda/6.5.14 $CUDA_MODULE_LIST $CUDA_BUILD_LIST $KOKKOS_PATH/config/nvcc_wrapper $CUDA_WARNING_FLAGS"
+               "cuda/7.0.28 $CUDA_MODULE_LIST $CUDA_BUILD_LIST $KOKKOS_PATH/config/nvcc_wrapper $CUDA_WARNING_FLAGS"
+               "cuda/7.5.18 $CUDA_MODULE_LIST $CUDA_BUILD_LIST $KOKKOS_PATH/config/nvcc_wrapper $CUDA_WARNING_FLAGS"
+    )
+
+elif [ "$MACHINE" = "white" ]; then
+    SKIP_HWLOC=True
+
+    BASE_MODULE_LIST="<COMPILER_NAME>/<COMPILER_VERSION>"
```
Comment (jgfouca) [inline config/test_all_sandia]: Need additional modules?

**Comment by jgfouca** [comment]
Hi @crtrott , have you had a chance to think about this?

**Comment by jgfouca** [comment]
@crtrott , @hcedwar , I'd love to hear from someone regarding this PR.

**Comment by crtrott** [comment]
I also updated the file to not test known broken configurations, as well as use the correct modules on white (the head-node has non-valid modules).

**Pull request merged**